exporter/datadogexporter/integrationtest/integration_test.go

```diff
@@ -618,3 +645,58 @@ func testIntegrationHostMetrics(t *testing.T, expectedMetrics map[string]struct{
 		}
 	}
 }
+
+func testIntegrationHostMetricsForSerializer(t *testing.T, expectedMetrics map[string]struct{}) {
+	// 1. Set up mock Datadog server
+	seriesRec := &testutil.HTTPRequestRecorderWithChan{Pattern: testutil.MetricV2Endpoint, ReqChan: make(chan []byte, 100)}
+	server := testutil.DatadogServerMock(seriesRec.HandlerFunc)
+	defer server.Close()
+	t.Setenv("SERVER_URL", server.URL)
+
+	// 2. Start in-process collector
+	factories := getIntegrationTestComponents(t)
+	app := getIntegrationTestCollector(t, "integration_test_host_metrics_config.yaml", factories)
+	go func() {
+		assert.NoError(t, app.Run(context.Background()))
+	}()
+	defer app.Shutdown()
+
+	waitForReadiness(app)
+
+	// 3. Validate host metrics in DD and/or OTel conventions are sent to the mock server
+	// See https://docs.datadoghq.com/opentelemetry/integrations/host_metrics/?tab=host
+	metricMap := make(map[string]series)
+	for len(metricMap) < len(expectedMetrics) {
+		select {
+		case metricsBytes := <-seriesRec.ReqChan:
+			zr, err := zlib.NewReader(bytes.NewReader(metricsBytes))
+			assert.NoError(t, err)
+			pl := new(gogen.MetricPayload)
+			b, err := io.ReadAll(zr)
+			assert.NoError(t, err)
+
+			if err = pl.Unmarshal(b); err != nil {
+				t.Fatalf("Failed to unmarshal payload: %v", err)
+			}
+			assert.NoError(t, err)
+			for _, s := range pl.GetSeries() {
+				if _, ok := expectedMetrics[s.Metric]; ok {
+					var points []point
+					for _, p := range s.Points {
+						points = append(points, point{
+							Timestamp: int(p.GetTimestamp()),
+							Value:     p.GetValue(),
+						})
+					}
+					metricMap[s.Metric] = series{
+						Metric: s.Metric,
+						Points: points,
+					}
+				}
+			}
```

Comment:
nit: I would just move these to a helper function and reuse in `testIntegrationHostMetrics`, e.g.
```
func testIntegrationHostMetrics(t *testing.T, expectedMetrics map[string]struct{}, useSerializer bool) {
...
  case metricsBytes := <-seriesRec.ReqChan:
    if useSerializer {
      metricMap = seriesFromSerializer
    } else ...
}

func seriesFromSerializer() map[string]series {
  zr, err := zlib.NewReader(bytes.NewReader(metricsBytes))
  ...
  return metricMap
}

func seriesFromAPIClient() map[string]series {
  gz := getGzipReader(t, metricsBytes)
  ...
  return metricMap
}
```